### PR TITLE
darwin nat implementation

### PR DESCRIPTION
**Pull request by zolia**
Client still has to be on different machine. 
This is due to 10.8.0.0/24 subnet clash. That is server wants to tunnel it through utun1, but client would need to tunnel it through utun2, but cannot do it since route already exists.

### CODE REVIEW

File: nat/service_pfctl.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+}
+
+func (service *servicePFCtl) enableRules() error {
+	for _, rule := range service.rules {
+		iface, err := ifaceByAddress(rule.TargetIP)
+		if err != nil {
+			return err
+		}
+		natRule := fmt.Sprintf("nat on %v inet from %v to any -> %v", iface, rule.SourceAddress, rule.TargetIP)
+		arguments := fmt.Sprintf("echo \"%v\" | /sbin/pfctl -vEf -", natRule)
```
Comment (Waldz) [inline nat/service_pfctl.go]: Use qoute ` to avoid \\" escaping

File: nat/service_pfctl.go [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+			}
+		}
+
+		log.Info(natLogPrefix, "NAT from '", rule.SourceAddress, "' to IP: ", rule.TargetIP)
+	}
+	return nil
+}
+
+func (service *servicePFCtl) disableRules() {
+	arguments := fmt.Sprintf("/sbin/pfctl -F nat")
```
Comment (Waldz) [inline nat/service_pfctl.go]: `Sprintf ` is redundant, because there aren't variables

File: nat/service_pfctl.go [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+	cmd := exec.Command(
+		"sh",
+		"-c",
+		arguments,
```
Comment (Waldz) [inline nat/service_pfctl.go]: Looks like sub-command quoting is missing, is not it?

File: nat/service_pfctl.go [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+		if err != nil {
+			return "", err
+		}
+		for _, address := range addresses {
+			if address.(*net.IPNet).IP.String() == ipAddress {
+				return ifi.Name, nil
+			}
+		}
+	}
+	return "undefined", nil
```
Comment (Waldz) [inline nat/service_pfctl.go]: Is not it error?

File: nat/service_pfctl.go [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+	log.Info(natLogPrefix, "IP forwarding enabled")
+	return nil
+}
+
+func (service *servicePFCtl) disableIPForwarding() {
+	if service.forward {
+		return
+	}
+
+	cmd := utils.SplitCommand("/usr/sbin/sysctl", "-w net.inet.ip.forwarding=0")
```
Comment (Waldz) [inline nat/service_pfctl.go]: Looks like duplicated code, could be DRY'ied

File: nat/service_pfctl.go [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+		cmd := exec.Command(
+			"sh",
+			"-c",
+			arguments,
+		)
+
+		if output, err := cmd.CombinedOutput(); err != nil {
+			if !strings.Contains(string(output), natRule) {
+				log.Warn("Failed to create pfctl rule: ", cmd.Args, " Returned exit error: ", err.Error(), " Cmd output: ", string(output))
+				log.Flush()
```
Comment (Waldz) [inline nat/service_pfctl.go]: What is so special about your logs, that you flush it?

File: nat/service_pfctl.go (hunk 1 above)
Comment (Waldz) [reply nat/service_pfctl.go]: Could could be just `/sbin/pfctl -vE -f %s`?

File: nat/service_pfctl.go (hunk 5 above)
Comment (zolia) [reply nat/service_pfctl.go]: its not, different path of command and different arguments.

File: nat/service_pfctl.go (hunk 1 above)
Comment (zolia) [reply nat/service_pfctl.go]: no, its input from stdin (-)

File: nat/service_pfctl.go (hunk 6 above)
Comment (tadovas) [reply nat/service_pfctl.go]: There is no need to flush logs on every step. Flushing is done on main package just before program exiting

**Pull request merged**